src/Orc.Automation/ScenarioManagement/ScenarioManager.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace Orc.Automation.ScenarioManagement;
+
+using System;
+
+public static class ScenarioManager
+{
+    private static AutomationScenario? _currentScenario;
+
+    public static AutomationScenario? CurrentScenario => _currentScenario;
+
+    public static bool IsRecordingStepsSuspended { get; set; }
```

Comment:
Double check whether setter should be public

Reply:
yes, it used by Attribute...